zerver/views/realm_logo.py

```diff
@@ -2,52 +2,77 @@
+from zerver.lib.realm_night_logo import realm_night_logo_url
 from zerver.lib.response import json_error, json_success
 from zerver.lib.upload import upload_logo_image
 from zerver.models import Realm, UserProfile
 
 
 @require_realm_admin
-def upload_logo(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
+@has_request_variables
+def upload_logo(request: HttpRequest, user_profile: UserProfile,
+                night: bool=REQ(validator=check_bool)) -> HttpResponse:
     if user_profile.realm.plan_type == Realm.LIMITED:
         return json_error(_("Feature unavailable on your current plan."))
 
     if len(request.FILES) != 1:
         return json_error(_("You must upload exactly one logo."))
-
     logo_file = list(request.FILES.values())[0]
     if ((settings.MAX_LOGO_FILE_SIZE * 1024 * 1024) < logo_file.size):
         return json_error(_("Uploaded file is larger than the allowed limit of %s MB") % (
             settings.MAX_LOGO_FILE_SIZE))
-    upload_logo_image(logo_file, user_profile)
-    do_change_logo_source(user_profile.realm, user_profile.realm.LOGO_UPLOADED)
-    logo_url = realm_logo_url(user_profile.realm)
-
-    json_result = dict(
-        logo_url=logo_url
-    )
+    if night:
+        upload_logo_image(logo_file, user_profile, night)
+        do_change_logo_source(user_profile.realm, user_profile.realm.NIGHT_LOGO_UPLOADED, night)
+        logo_url = realm_night_logo_url(user_profile.realm)
+        json_result = dict(
+            night_logo_url=logo_url
+        )
+    else:
+        upload_logo_image(logo_file, user_profile, night)
+        do_change_logo_source(user_profile.realm, user_profile.realm.LOGO_UPLOADED, night)
+        logo_url = realm_logo_url(user_profile.realm)
+        json_result = dict(
+            logo_url=logo_url
+        )
     return json_success(json_result)
 
-
 @require_realm_admin
-def delete_logo_backend(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
+@has_request_variables
+def delete_logo_backend(request: HttpRequest, user_profile: UserProfile,
+                        night: bool=REQ(validator=check_bool)) -> HttpResponse:
     # We don't actually delete the logo because it might still
     # be needed if the URL was cached and it is rewrited
     # in any case after next update.
-    do_change_logo_source(user_profile.realm, user_profile.realm.LOGO_DEFAULT)
-    default_url = realm_logo_url(user_profile.realm)
-    json_result = dict(
-        logo_url=default_url
-    )
+    if night:
+        do_change_logo_source(user_profile.realm, user_profile.realm.NIGHT_LOGO_DEFAULT, night)
+        default_url = realm_night_logo_url(user_profile.realm)
+        json_result = dict(
+            night_logo_url=default_url
+        )
+    else:
+        do_change_logo_source(user_profile.realm, user_profile.realm.LOGO_DEFAULT, night)
+        default_url = realm_logo_url(user_profile.realm)
+        json_result = dict(
+            logo_url=default_url
+        )
```

Comment:
See the below comment.  I think with a similar technique, we could shrink the `if/else` here to just be around a variable `default_logo = `...

Reply:
Actually, I think we can also have the same constants `LOGO_DEFAULT` -- no need for separate `NIGHT_LOGO_*` ones.  So in fact, it's only the `json_result` piece that needs a conditional.

And the same reasoning applies to the upload code path -- you can move most of the code out of the if/else block.